### DISCUSSION TITLE
Any plans to add Circle, Polygon or Polyline in the library as components?

### DISCUSSION THREAD

**Discussion by sllonch** [visgl/react-google-maps#429] [General]
I appreciate the work and effort you put on having these components available in this [MR](https://github.com/visgl/react-google-maps/pull/178) but I'm wondering if there are any plans to include them in the library in further releases.

Cheers!

**Comment by usefulthink** [comment]
Maybe, but there's nothing planned yet. 

Is there a specific reason why you'd prefer this to be bundled with the library (as opposed to using the component we have in our examples)?

**Comment by Zeden19** [comment]
I know this is really old but I stumbled across this whilst transitioning one of my projects over to this library.

If I remember right, the justification to have these elements be copy pasted was that it wasn't a problem in ShadCN. My issue with that is the fact that ShadCN is entirely based around copy and pasting elements into your own project, whilst this project has both library components and copy pasted components. (As a side note, ShadCN uses CLI to copy paste components now anyways)

In my opinion it feels strange to have some elements (like AdvancedMarker) bundled with the library and some elements be copy pasted. When I was implementing Polyline into my project I just assumed it would be bundled because other elements were also bundled.

I'm not sure if this is a common pattern in other projects like this, but I thought it was just worth sharing my experience.

**Reply by usefulthink** [reply]
First of all, you will be interested in PR #917 :)

I'm not aware of other projects doing this, but to me it felt like a good idea to address the quality vs time invested tradeoff when we created this library. 

If we had tried to add support for all the features of the maps API at once (or even any larger percentage), it would have negatively impacted the quality in several ways. The amount of time I and the other maintainers can spend on the project is limited. More functionality means there will be more bugs or unexpected behaviour, and the time required to write documentation, fix bugs, answer questions and so on increases. If that time isn't available, we end up with unsolved issues and unanswered questions and an overall worse quality for the library. 

However, we had always planned to add full support for the features that are most useful to our users, which is what we are doing now by adding first-class support for geometry primitives and 3d-maps. (I think both of which will be in the 1.8 release.)
